From 147d4245a0255813bf2d941763f5b3b51fab40af Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <shzam@sdf.org>
Date: Wed, 12 Jul 2023 13:33:18 +1000
Subject: [PATCH 1/2] Copy over some macros from libavutil/version.h

Wihtout these the structure included in libraries other than libavutil
is binary incompatible with the one in libavutil. This causes crashes.
---
 libavutil/version.h.in | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libavutil/version.h.in b/libavutil/version.h.in
index 99c6af6732..c7ad155c30 100644
--- a/libavutil/version.h.in
+++ b/libavutil/version.h.in
@@ -129,6 +129,18 @@
 #ifndef FF_API_PSEUDOPAL
 #define FF_API_PSEUDOPAL                (LIBAVUTIL_VERSION_MAJOR < 57)
 #endif
+#ifndef FF_API_CHILD_CLASS_NEXT
+#define FF_API_CHILD_CLASS_NEXT         (LIBAVUTIL_VERSION_MAJOR < 57)
+#endif
+#ifndef FF_API_BUFFER_SIZE_T
+#define FF_API_BUFFER_SIZE_T            (LIBAVUTIL_VERSION_MAJOR < 57)
+#endif
+#ifndef FF_API_D2STR
+#define FF_API_D2STR                    (LIBAVUTIL_VERSION_MAJOR < 58)
+#endif
+#ifndef FF_API_DECLARE_ALIGNED
+#define FF_API_DECLARE_ALIGNED          (LIBAVUTIL_VERSION_MAJOR < 58)
+#endif
 
 
 /**
-- 
2.40.1

